iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java

```diff
@@ -194,10 +197,18 @@ private void analyseFile(InputFileContext inputFileContext) {
         } catch (RuntimeException e) {
           inputFileContext.reportAnalysisError(e.getMessage(), null);
           LOG.error("Cannot analyse '" + inputFile +"': " + e.getMessage(), e);
+
+          interruptOnFailFast(inputFileContext.sensorContext, inputFile, e);
         }
       }
     }
 
+    private void interruptOnFailFast(SensorContext context, InputFile inputFile, Exception e) {
+      if (context.config().getBoolean(FAIL_FAST_PROPERTY_NAME).orElse(false)) {
```

Comment:
Minor. The value can be read once and cached

Reply:
Fair point. I thought about the same. But how often we should have exceptions :see_no_evil: ?